### DISCUSSION TITLE
Basic question on usage and IP addresses of tuya devices

### DISCUSSION THREAD

**Discussion by zenonmb** [jasonacox/tinytuya#545] [Q&A]
This tinytuya alows local connection to the devices, but first needs to pair devices using mobile App and for that it needs internet access . So,for initial setup sensors need internet access . Then once setup it will need no more . Am I right?
Also noticed we need a defined IP address and using DHCP that might change over time. Can we still use tinytuya? or weneed fixed ipp add schema?

**Comment by RYoder97** [comment]
Hi there!

So what I did was disconnect my coax cable from my modem/router combo, and tried to use tinytuya. It still worked! And from what I understand, so long as you NEVER remove the devices and re-pair them via the Tuya app, the local keys will never change.

It sucks that you need the initial connection, but I think if you do enough digging it looks like there may be ways to get the key without the cloud.

**Comment by uzlonewolf** [comment]
It is very much device dependent.  Some devices work just fine without internet, others kinda work but stop updating their status, and some won't work at all.  Unfortunately the only way to find out is to try it.

If you are using dynamic IPs then you can use the auto-IP function, however this will result in a long delay (up to 5 seconds) for the very first connection.
